extensions/amp-access-scroll/0.1/scroll-impl.js

```diff
@@ -133,3 +164,35 @@ class ScrollElement {
         });
   }
 }
+
+/**
+ * Add analytics for Scroll to page.
+ * @param {!Element} parentElement
+ * @param {!JsonObject} vendorConfig
+ */
+function addAnalytics(parentElement, vendorConfig) {
```

Comment:
Ha, sounds good. FWIW, the merged version is definitely messier, so I think if anything it would make more sense to make two functions and then just factor out some of the shared pieces.